src/vmprof_main.h

```diff
@@ -175,11 +179,43 @@ static PY_THREAD_STATE_T * _get_pystate_for_this_thread(void) {
 }
 #endif
 
+#ifdef VMPROF_LINUX
+static void broadcast_signal_for_threads(pid_t pid)
```

Comment:
Yes, true the kernel delivers that info. If they calls are not signal safe, the program most likely freezes or crashes.
We could think about hooking into thread creation, that would register/unregister the thread id. Though a solution asking the kernel would be much better.